### PR TITLE
Wrapper for UserProfile extension

### PR DESCRIPTION
**Pull request by praveek**
https://github.com/adobe/aepsdk_flutter/issues/38 -  Add wrapper for UserProfile extension

### CODE REVIEW

**Comment by cacheung** [comment]
We need to add userprofile to the main readme.
https://github.com/adobe/aepsdk_flutter#initializing
Also need to update the initializing link pointing to developer.com.

File: plugins/flutter_aepuserprofile/android/gradlew [hunk 1]
```diff
@@ -0,0 +1,234 @@
+#!/bin/sh
+
```
Comment (cacheung) [inline plugins/flutter_aepuserprofile/android/gradlew:2]: I don't see these three files in other extensions:  gradlew, gradlew.bat, gradle/wrapper/gradle-wrapper.jar , should we remove them?

File: plugins/flutter_aepuserprofile/ios/Classes/FlutterAEPUserProfilePlugin.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
```
Comment (cacheung) [inline plugins/flutter_aepuserprofile/ios/Classes/FlutterAEPUserProfilePlugin.h:2]: 2023?

File: plugins/flutter_aepuserprofile/ios/Classes/FlutterAEPUserProfilePlugin.m [hunk 3]
```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2020 Adobe. All rights reserved.
```
Comment (cacheung) [inline plugins/flutter_aepuserprofile/ios/Classes/FlutterAEPUserProfilePlugin.m:2]: 2023?

File: plugins/flutter_aepuserprofile/pubspec.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
+name: flutter_aepuserprofile
+description: A new Flutter plugin project.
```
Comment (cacheung) [inline plugins/flutter_aepuserprofile/pubspec.yaml:2]: Have a description specific to user profile.

File: plugins/flutter_aepuserprofile/pubspec.yaml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+version: 1.0.0
+homepage:
+
```
Comment (cacheung) [inline plugins/flutter_aepuserprofile/pubspec.yaml:5]: Add
homepage: https://developer.adobe.com/client-sdks
repository: https://github.com/adobe/aepsdk-flutter/tree/main/plugins/flutter_aepuserprofile

File: plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return null;
+      });
+    });
+
+    test('invokes correct method', () async {
```
Comment (cdhoffmann) [inline plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart:77]: Maybe we should add a test case for passing empty strings. Does it let you pass nil or no?

File: plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
```
Comment (cdhoffmann) [inline plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart:100]: We should also test passing incorrect arguments potentially.

File: plugins/flutter_aepuserprofile/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepuserprofile/FlutterAEPUserProfilePlugin.java [hunk 8]
```diff
@@ -0,0 +1,126 @@
...
+
+  @Override
+  public void onMethodCall(MethodCall call, Result result) {
+    if ("extensionVersion".equals(call.method)) {
+      result.success(UserProfile.extensionVersion());
+    } else if ("getUserAttributes".equals((call.method))) {
+      handleGetUserAttributes(call.arguments, result);
+    } else if ("removeUserAttributes".equals(call.method)) {
+      handleRemoveUserAttributes(call.arguments);
+      result.success(null);
```
Comment (cdhoffmann) [inline plugins/flutter_aepuserprofile/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepuserprofile/FlutterAEPUserProfilePlugin.java:67]: It's possible for the result to be success even if the arguments are invalid. Is that expected?

**Review by cdhoffmann — CHANGES_REQUESTED** [review]

File: plugins/flutter_aepuserprofile/android/gradlew (hunk 1 above)
Comment (praveek) [reply plugins/flutter_aepuserprofile/android/gradlew]: It should be fine to remove those files with the way we have set up sample app.

File: plugins/flutter_aepuserprofile/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepuserprofile/FlutterAEPUserProfilePlugin.java (hunk 8 above)
Comment (praveek) [reply plugins/flutter_aepuserprofile/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepuserprofile/FlutterAEPUserProfilePlugin.java:67]: Dart only allows passing a `List<String>` to this API. This call completes the future as even our native extension API does not return any value to denote if the call has succeeded.

File: plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart (hunk 7 above)
Comment (praveek) [reply plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart:100]: There will be a compilation error if we try to pass incorrect arguments.

File: plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart (hunk 6 above)
Comment (praveek) [reply plugins/flutter_aepuserprofile/test/flutter_aepuserprofile_test.dart:77]: It will let us pass empty strings. The extension logic takes care of such checks and we also don't do such checks in our native APIs.

**Review by cdhoffmann — APPROVED** [review]

**Review by cacheung — APPROVED** [review]
